docs/algorithms/sig/falcon.md

```diff
@@ -7,16 +7,18 @@
 - **Authors' website**: https://falcon-sign.info
 - **Specification version**: 20211101.
 - **Primary Source**<a name="primary-source"></a>:
-  - **Source**: https://github.com/PQClean/PQClean/commit/0657749a785db30e7f49e9435452cb042edb1852
+  - **Source**: https://github.com/PQClean/PQClean/commit/7707d1bcc8ae7f9ffd296dd13b1d76d2767d14f8
   - **Implementation license (SPDX-Identifier)**: MIT
 
 
 ## Parameter set summary
 
-|  Parameter set  | Parameter set alias   | Security model   |   Claimed NIST Level |   Public key size (bytes) |   Secret key size (bytes) |   Signature size (bytes) |
-|:---------------:|:----------------------|:-----------------|---------------------:|--------------------------:|--------------------------:|-------------------------:|
-|   Falcon-512    | NA                    | EUF-CMA          |                    1 |                       897 |                      1281 |                      666 |
-|   Falcon-1024   | NA                    | EUF-CMA          |                    5 |                      1793 |                      2305 |                     1280 |
+|    Parameter set     | Parameter set alias   | Security model   |   Claimed NIST Level |   Public key size (bytes) |   Secret key size (bytes) |   Signature size (bytes) |
```

Comment:
This change confuses me: The length of the artifacts created by the (currently named) "Falcon" algs change while the (new, padded) algs take on those lengths: Is this intentional? Isn't it surprising then that KATs for the ("original") Falcon algs don't change?

Reply:
The only length which actually changes is the signature length. The "original" / "currently named" Falcon algorithm produces variable-length signatures, so this length is only an upper bound. We had (incorrectly, per the spec) been using the size of the fixed-length "PADDED" format as an upper bound previously.

The behaviour of the "original" / "currently named" algs _does_ change with regard to KATs. In fact, the behaviour changes precisely so that Falcon-1024 now produces the correct KATs. Previously, our implementation had been out of sync with the KATs: see #1561.

Reply:
> The behaviour of the "original" / "currently named" algs does change with regard to KATs.

Aha. That then has an implication for https://github.com/open-quantum-safe/oqs-provider/pull/362... (new OID necessary for Falcon-512/1024).

Reply:
> > The behaviour of the "original" / "currently named" algs does change with regard to KATs.
> 
> Aha. That then has an implication for [open-quantum-safe/oqs-provider#362](https://github.com/open-quantum-safe/oqs-provider/pull/362)... (new OID necessary for Falcon-512/1024).

The latest updates to https://github.com/open-quantum-safe/oqs-provider/pull/362 assign new OIDs for Falcon-512 and Falcon-1024. This will of course require another update once the new names are integrated, but from an OID standpoint I think it's OK now.

Reply:
> but from an OID standpoint I think it's OK now.

I tend to agree just looking at the YML. But when checking the resultant documentation I'm a bit confused (see https://github.com/open-quantum-safe/oqs-provider/pull/362#discussion_r1513177365). Is this the consequence of getting the KATs right -- did we indeed not change the underlying code? I thought we did -- wouldn't then a different algorithm version ID be prudent?